craft_providers/multipass/multipass_instance.py

```diff
@@ -393,10 +394,29 @@ def push_file(self, *, source: pathlib.Path, destination: pathlib.PurePath) -> N
                 f"Directory not found: {str(destination.parent.as_posix())!r}"
             )
 
-        self._multipass.transfer(
-            source=str(source),
-            destination=f"{self.name}:{destination.as_posix()}",
-        )
+        try:
+            tmp_file_path = self._create_temp_file()
```

Comment:
this method assumes an ubuntu user; we should probably change our methods to have an execute_run argument to not sudo.

Reply:
Added a new issue: https://github.com/canonical/craft-providers/issues/281